test/internal/kubernetes/utils.go

```diff
@@ -466,10 +466,37 @@ func MustRestartDaemonset(ctx context.Context, clientset *kubernetes.Clientset,
 		ds.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
 	}
 
+	gen := ds.Status.ObservedGeneration
+	log.Printf("Current generation is %v", gen)
 	ds.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
 
 	_, err = clientset.AppsV1().DaemonSets(namespace).Update(ctx, ds, metav1.UpdateOptions{})
-	return errors.Wrapf(err, "failed to update ds %s", daemonsetName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update ds %s", daemonsetName)
+	}
+	checkDaemonsetGenerationFn := func() error {
+		ds, err := clientset.AppsV1().DaemonSets(namespace).Get(ctx, daemonsetName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "could not get daemonset %s", daemonsetName)
+		}
+
+		if ds.Status.ObservedGeneration < gen {
```

Comment:
Technically if generation is less then observed then a restart has occurred, but not an intended one.